tests/installation/npmTest.ts

```diff
@@ -77,25 +63,88 @@ _expect.extend({
+  tsc: (...argsAndOrOptions: ArgsOrOptions) => Promise<string>;
+  registry: Registry;
 };
 
 export const test = _test
     .extend<CommonFixtures, CommonWorkerFixtures>(commonFixtures)
-    .extend<NPMTestFixtures>({
+    .extend<NPMTestFixtures & NPMTestOptions>({
+      useRealCDN: [false, { option: true }],
+      _cdnProxyEnv: async ({ useRealCDN }, use) => {
+        if (useRealCDN) {
+          await use({});
+          return;
+        }
+
+        const server = createHttpServer(async (request: http.IncomingMessage, response: http.ServerResponse) => {
+          const requestedPath = url.parse(request.url!).path;
+          const cachedPath = path.join(BROWSERS_CACHE_DIR, calculateSha1(requestedPath));
+          const cachedPathMetaInfo = cachedPath + '.metainfo';
+
+          if (!fs.existsSync(cachedPath)) {
+            const realUrl = 'https://playwright.azureedge.net' + requestedPath;
+            debug(`[cdn proxy] downloading ${realUrl} headers=${JSON.stringify(request.headers)}`);
+            const headers = { ...request.headers };
+            delete headers['host'];
+            const options = {
+              ...url.parse(realUrl),
+              method: request.method,
+              headers,
+            };
+            const factory = options.protocol === 'https:' ? https : http;
+            let doneCallback = () => {};
+            const donePromise = new Promise<void>(f => doneCallback = () => {
+              debug(`[cdn proxy] downloading ${realUrl} finished`);
+              f();
+            });
+            const realRequest = factory.request(options, (realResponse: http.IncomingMessage) => {
+              const metaInfo = {
+                statusCode: realResponse.statusCode,
+                statusMessage: realResponse.statusMessage || '',
+                headers: realResponse.headers || {},
+              };
+              debug(`[cdn proxy] downloading ${realUrl} statusCode=${realResponse.statusCode}`);
+              fs.mkdirSync(path.dirname(cachedPathMetaInfo), { recursive: true });
+              fs.writeFileSync(cachedPathMetaInfo, JSON.stringify(metaInfo));
+              realResponse.pipe(fs.createWriteStream(cachedPath)).on('close', doneCallback).on('finish', doneCallback).on('error', doneCallback);
+            });
+            request.pipe(realRequest);
+            await donePromise;
+          }
+
+          const metaInfo = JSON.parse(fs.readFileSync(cachedPathMetaInfo, 'utf-8'));
+          response.writeHead(metaInfo.statusCode, metaInfo.statusMessage, metaInfo.headers);
+          const done = () => {
+            debug(`[cdn proxy] serving ${request.url!} finished`);
+            response.end();
+          };
+          fs.createReadStream(cachedPath).pipe(response).on('close', done).on('error', done);
+          debug(`[cdn proxy] serving ${request.url!} from cached ${cachedPath}`);
+        });
+
+        const PORT = 15689;
```

Comment:
shall we have PORT 0 here? Alternatively, can we start the CDN in global setup once for everybody?